src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSCloud.java

```diff
@@ -363,21 +396,32 @@ public void setJenkinsUrl(String jenkinsUrl) {
         }
     }
 
+
     /**
      * Add a dynamic task template. Won't be displayed in UI, and persisted separately from the cloud instance.
-     * @param t the template to add
+     * @param template the template to add
+     * @return the task definition created from the template
      */
-    public void addDynamicTemplate(ECSTaskTemplate t) {
-        TaskTemplateMap.get().addTemplate(this, t);
+    public TaskDefinition  addDynamicTemplate(ECSTaskTemplate template) {
```

Comment:
I believe this change broke the ability to use task definition overrides in pipeline jobs. On 1.34 I'm seeing

`
com.amazonaws.services.ecs.model.ClientException: Container.image should not be null or empty. (Service: AmazonECS; Status Code: 400; Error Code: ClientException; Request ID: iRedactedThis)`